### ISSUE TITLE
On windows, problems with file with accents

### ISSUE THREAD

**Issue by vedard** [supermihi/pytaglib#31]
Windows 10
Python 3.5.3
pytaglib 1.4.0 (install with pip)

File with accent cause a OSError, other file work correctly. On linux there is no problem with the same file
```
C:\Users\vincent>python
Python 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import taglib, os
>>> song = r'D:\Music\iTunes\iTunes Media\Music\Beyond Creation\The Aura\06 Le Détenteur.m4a'
>>> song
'D:\\Music\\iTunes\\iTunes Media\\Music\\Beyond Creation\\The Aura\\06 Le Détenteur.m4a'
>>> os.path.exists(song)
True
>>> taglib.File(song)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/taglib.pyx", line 82, in taglib.File.__cinit__ (src\taglib.cpp:1597)
OSError: Could not read file D:\Music\iTunes\iTunes Media\Music\Beyond Creation\The Aura\06 Le Détenteur.m4a
>>> song2 = r'D:\Music\iTunes\iTunes Media\Music\Beyond Creation\The Aura\05 Injustice Revealed.m4a'
>>> taglib.File(song2)
File('D:\Music\iTunes\iTunes Media\Music\Beyond Creation\The Aura\05 Injustice Revealed.m4a')
>>>
```

**Comment by supermihi** [comment]
Sorry for the delayed response. This issue sounds like #28, but apparently you are using the newest version already ...thanks for reporting,  I'll have a look ASAP.

**Comment by ngc92** [comment]
I can confirm that the problem also exists in pytaglib 1.4.1 / window 7

**Comment by jeroonk** [comment]
The culprit here is this conditional Cython compilation: https://github.com/supermihi/pytaglib/blob/50b04ab11c36c0b67373d514586919240530d5ae/src/taglib.pyx#L75-L80 The cythonized taglib.cpp included with the shipped version of pytaglib corresponds to a non-windows version, as can be seen by it taking the ELSE branch here: https://github.com/supermihi/pytaglib/blob/50b04ab11c36c0b67373d514586919240530d5ae/src/taglib.cpp#L1718-L1725

When I delete taglib.cpp (otherwise setup.py doesn't seem to overwrite it, even with `--cython`) and rebuild the extension on Windows, Cython takes the IF branch instead. This results in it generating `Py_UNICODE const *__pyx_t_5` instead of `char const *__pyx_t_5` here: https://github.com/supermihi/pytaglib/blob/50b04ab11c36c0b67373d514586919240530d5ae/src/taglib.cpp#L1645-L1647 And generating `__Pyx_PyUnicode_AsUnicode` instead of `__Pyx_PyBytes_AsString` here: https://github.com/supermihi/pytaglib/blob/50b04ab11c36c0b67373d514586919240530d5ae/src/taglib.cpp#L1728-L1730

The recompiled extension has no problem with special characters in filenames.
I am not sure what the best approach for solving this is:
- Somehow make taglib.pyx emit actual `#ifdef _WIN32` (cpp-)compiler directives to compile against both Windows-style `wchar_t*` and unicode `char*` versions of `TagLib::FileRef::create`. Would probably require some ugly hacks, (`create` has a conditional cdef in ctypes.pxd too). There is [probably not going to be](https://groups.google.com/forum/#!topic/cython-users/HASNraxsGGw) any official Cython support for something like this.
- Remove taglib.cpp and add Cython back as a dependency for everyone.
- Somehow force setup.py to re-cythonize taglib.cpp on Windows and add Cython as a dependency for Windows users not using the binaries.
- Ship a binary version for all Python versions on Windows

**Comment by supermihi** [comment]
Thanks a lot for investigating this!!!

I prefer the two last options as a "fix". Using pytaglib on Windows without the binary backages is complex anyway, it doesn't really matter to add Cython as another dependency. ;-)
I'm going to prepare a fixed release as soon as I find the time (and a Windows machine to build the binary wheels for PyPI).

**Comment by supermihi** [comment]
Fixed in v1.4.2
